### ISSUE TITLE
[sdk] kfp.deprecated fails to import

### ISSUE THREAD

**Issue by hmc-cs-mdrissi** [kubeflow/pipelines#9803]
### Environment

*  KFP version: kfp 2.1.2

### Steps to reproduce
```
>>> import kfp.deprecated
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/pa-loaner/Snapchat/Dev/.venvs/bento/lib/python3.9/site-packages/kfp/deprecated/__init__.py", line 19, in <module>
    from . import dsl
  File "/Users/pa-loaner/Snapchat/Dev/.venvs/bento/lib/python3.9/site-packages/kfp/deprecated/dsl/__init__.py", line 16, in <module>
    from ._pipeline import Pipeline, PipelineExecutionMode, pipeline, get_pipeline_conf, PipelineConf
  File "/Users/pa-loaner/Snapchat/Dev/.venvs/bento/lib/python3.9/site-packages/kfp/deprecated/dsl/_pipeline.py", line 21, in <module>
    from kfp.deprecated.dsl import _component_bridge
  File "/Users/pa-loaner/Snapchat/Dev/.venvs/bento/lib/python3.9/site-packages/kfp/deprecated/dsl/_component_bridge.py", line 29, in <module>
    from kfp.deprecated.dsl import component_spec as dsl_component_spec
  File "/Users/pa-loaner/Snapchat/Dev/.venvs/bento/lib/python3.9/site-packages/kfp/deprecated/dsl/component_spec.py", line 21, in <module>
    from kfp.deprecated.dsl import type_utils
  File "/Users/pa-loaner/Snapchat/Dev/.venvs/bento/lib/python3.9/site-packages/kfp/deprecated/dsl/type_utils.py", line 19, in <module>
    from kfp.components.types import type_utils
ModuleNotFoundError: No module named 'kfp.components.types'
```

2.0.1/2.0.0 leads to a different incompatibility,

```
  File "/Users/pa-loaner/Snapchat/Dev/.venvs/bento/lib/python3.9/site-packages/kfp/deprecated/_client.py", line 97, in <module>
    class Client(object):
  File "/Users/pa-loaner/Snapchat/Dev/.venvs/bento/lib/python3.9/site-packages/kfp/deprecated/_client.py", line 396, in Client
    def get_kfp_healthz(self) -> kfp_server_api.ApiGetHealthzResponse:
AttributeError: module 'kfp_server_api' has no attribute 'ApiGetHealthzResponse'
```

ApiGetHealthzResponse is present in kfp-server-api <2.0, but gone in 2.0. Except kfp depends on >=2.0. So it's unclear to me kfp.deprecated is usable. I'd also expect some unit test that imports/uses kfp.deprecated. Are all those tests disabled?

Using kfp-server-api v1.8 + kfp2 importing kfp.deprecated also fails because kfp v2 apis have type hints that refer to kfp-server-api v2 like,

```
  File "/Users/pa-loaner/Snapchat/Dev/.venvs/bento/lib/python3.9/site-packages/kfp/client/__init__.py", line 20, in <module>
    from kfp.client.client import Client
  File "/Users/pa-loaner/Snapchat/Dev/.venvs/bento/lib/python3.9/site-packages/kfp/client/client.py", line 78, in <module>
    class _JobConfig:
  File "/Users/pa-loaner/Snapchat/Dev/.venvs/bento/lib/python3.9/site-packages/kfp/client/client.py", line 80, in _JobConfig
    pipeline_version_reference: kfp_server_api.V2beta1PipelineVersionReference
AttributeError: module 'kfp_server_api' has no attribute 'V2beta1PipelineVersionReference'
```

### Expected result

kfp.deprecated is still usable on current kfp..

### Materials and Reference

Impacted by this bug? Give it a 👍.

**Comment by chensun** [comment]
> Expected result
>
> kfp.deprecated is still usable on current kfp..

@hmc-cs-mdrissi `kfp.deprecated` is meant to be cleanup but we haven't done so. It's not expected for it to continue working inside kfp 2.0+. But the features are still available via kfp sdk 1.8.*.

Is there any reason you upgrade to KFP 2.0+ and still want to use `kfp.deprecated`?

**Comment by hmc-cs-mdrissi** [comment]
I was exploring it to try upgrading. For codebase that uses kfp in several spots it would be helpful to be able to migrate portion of it to true kfp v2 and keep other parts as kfp.deprecated instead of updating all at once.

I did end up swapping to latest 1.8 though and my need to go through kfp.deprecated is low.
